tests/test_settings.py

```diff
@@ -82,24 +94,55 @@ def test_code_deps_and_base_unequal__raises_error(code_deps_base):
         run_build_settings(args)
 
 
+path_options = {  # options (-> settings members) that interact with basepath
+    "--code": "code",
+    "--deps": "deps",
+}
+
+Item = TypeVar("Item")
+
+
+def subsets(
+    items: Set[Item],
+    min_size: int = 0,
+    max_size: int = sys.maxsize,
```

Comment:
Is this set to `sys.maxsize` and then readapted in the first line of the function because:
- It's an optional argument so it needs a default value
- We can not set the default value to `max(items)` in the function call?

Reply:
💯 correct. In Python the default arguments (`= 0`, and `= sys.maxsize` above) are evaluated only _once_ when the function is first parsed/created by the Python interpreter. So the default argument has no way to reflect/update based on the value of an argument in a later call.

The other, maybe more common, way to write this would be something like:

```python
def subsets(
    items: Set[Item],
    min_size: int = 0,
    max_size: int = None,
) -> Iterator[Set[Item]]:
    max_size = len(items) if max_size is None else min(max_size, len(items))
```

(Aside: This is also why you should typically never use _mutable_ default arguments in Python, because the lifetime of the default-valued argument is non-obvious: the _same_ object is reused across all calls, and changes made in earlier function calls will be reflected in later function calls.)